tests/test_harvester_global_prateb.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import os,sys
+from datetime import datetime
+from pathlib import Path 
+import pytest
+import yaml
+import glob
+import xarray as xr
+from netCDF4 import Dataset
+
+from score_hv import hv_registry
+from score_hv.harvester_base import harvest
+from score_hv.yaml_utils import YamlLoader
+from score_hv.harvesters.innov_netcdf import Region, InnovStatsCfg
+
+#TEST_DATA_FILE_NAME = 'bfg_19940101_mean_prateb_control.nc'
+TEST_DATA_FILE_NAMES = ['bfg_1994010100_fhr09_prateb_control.nc',
+                        'bfg_1994010106_fhr06_prateb_control.nc',
+                        'bfg_1994010106_fhr09_prateb_control.nc',
+                        'bfg_1994010112_fhr06_prateb_control.nc',
+                        'bfg_1994010112_fhr09_prateb_control.nc',
+                        'bfg_1994010118_fhr06_prateb_control.nc',
+                        'bfg_1994010118_fhr09_prateb_control.nc',
+                        'bfg_1994010200_fhr06_prateb_control.nc']
+DATA_DIR           = 'data'
+CONFIGS_DIR        = 'configs'
+PYTEST_CALLING_DIR = Path(__file__).parent.resolve()
+TEST_DATA_PATH     = os.path.join(PYTEST_CALLING_DIR, DATA_DIR)
+BFG_PATH           = [os.path.join(TEST_DATA_PATH, file_name) for file_name in TEST_DATA_FILE_NAMES] 
+
+VALID_CONFIG_DICT = {
+    'harvester_name': hv_registry.GLOBAL_BUCKET_PRECIP_AVE,
+    'filenames' : BFG_PATH,
+    'statistic': ['mean'],
+    'variable': ['prateb_ave']
+ }
+
+def test_global_mean():
+    """ The harvester returns a numpy 32 bit floating point number.
+        The test must cast the global mean value hard coded here to a 
+        numpy.float32. Otherwise the assert function fails.
+    """
+    print("in test get mean")
+    data1               = harvest(VALID_CONFIG_DICT)
+    global_mean         = np.float32(2.4695819e-05)
```

Comment:
this is a nicely written test, I appreciate the comment explaining why it's being cast, and it's good to check against a pre-done calculation. however, I would recommend either putting in the comment or have a hardcoded input dictionary within the test what the file being used is if you are using a hardcoded calculated mean value so that it's clear to other users where that value should be coming from and will prevent a test being broken due to a change in input files (such as if we switch to a smaller test file for example). just looking at the test, I can't be sure where this mean value should be coming from.

Reply:
It does help, however, this could still be improved because someone could still change the values in the TEST_DATA_FILE_NAMES dictionary and that wouldn't be understood/reflected in this test without additional changes

Reply:
added the filenames to the docstring